src/core/lib/slice/slice_intern.cc

```diff
@@ -69,62 +64,27 @@ typedef struct {
 static static_metadata_hash_ent
     static_metadata_hash[4 * GRPC_STATIC_MDSTR_COUNT];
 static uint32_t max_static_metadata_hash_probe;
-static uint32_t static_metadata_hash_values[GRPC_STATIC_MDSTR_COUNT];
+uint32_t static_metadata_hash_values[GRPC_STATIC_MDSTR_COUNT];
 
-static void interned_slice_ref(void* p) {
-  interned_slice_refcount* s = static_cast<interned_slice_refcount*>(p);
-  GPR_ASSERT(gpr_atm_no_barrier_fetch_add(&s->refcnt, 1) > 0);
-}
+namespace grpc_core {
 
-static void interned_slice_destroy(interned_slice_refcount* s) {
-  slice_shard* shard = &g_shards[SHARD_IDX(s->hash)];
-  gpr_mu_lock(&shard->mu);
-  GPR_ASSERT(0 == gpr_atm_no_barrier_load(&s->refcnt));
+void InternedSliceRefcount::DestroyInstance() {
+  slice_shard* shard = &g_shards[SHARD_IDX(this->hash)];
+  MutexLock lock(&shard->mu);
+  // TODO(arjunroy) Really need this method
+  // GPR_ASSERT(0 == gpr_atm_no_barrier_load(&this->refcnt));
   interned_slice_refcount** prev_next;
   interned_slice_refcount* cur;
-  for (prev_next = &shard->strs[TABLE_IDX(s->hash, shard->capacity)],
+  for (prev_next = &shard->strs[TABLE_IDX(this->hash, shard->capacity)],
```

Comment:
Any reason for `this->` ? This isn't in a template declaration for example

Reply:
Readability; since that class hasn't been converted to _ suffixes on member vars this makes it a bit more obvious.